test/unit/bootstrap.js

```diff
@@ -132,16 +133,39 @@ global.document = {
     },
     createElementNS: (_, type) => (global.document.createElement(type)),
     getElementsByTagName: () => [new DOMElement()],
+    events: new Map(),
 };
 
+global.document.addEventListener = (event, cb) => { global.document.events.set(event, cb); };
+global.document.removeEventListener = () => {};
+global.document.emitEvent = (event, params) => {
+    const callback = global.document.events.get(event);
+    if (callback) {
+        return callback(params);
+    }
+};
 global.document.documentElement = global.document.createElement();
+global.document.body = new DOMElement();
+
+global.XRRigidTransform = () => {};
 
 class Renderer {
     constructor() {
         this.domElement = new DOMElement();
         this.domElement.parentElement = new DOMElement();
         this.domElement.parentElement.appendChild(this.domElement);
 
+        this.xr = new DOMElement();
```

Comment:
Why does `xr` need to be a `DomElement` instead of a simple `{}` object ?

Reply:
Was used because `DOMElement` already provide a listener implementation.
Added an `EventDispatcher` class to mimic this behavior (since `THREE.WebXRManager` extends it) and replaced it.